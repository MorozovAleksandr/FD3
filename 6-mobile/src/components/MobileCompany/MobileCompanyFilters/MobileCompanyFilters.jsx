import React from "react";
import { myEvents } from '../../../event';

class MobileCompanyFilters extends React.PureComponent {

    onClickFilter = (e) => {
        myEvents.emit('EfilteringClients', +e.target.dataset.mode);
    }

    render() {
        console.log('render MobileCompanyFilters');
        return (
            <div>
                <button data-mode={0} onClick={this.onClickFilter} type="button" className="btn btn-secondary btn-space">Все</button>
                <button data-mode={1} onClick={this.onClickFilter} type="button" className="btn btn-secondary btn-space">Активные</button>
                <button data-mode={2} onClick={this.onClickFilter} type="button" className="btn btn-secondary btn-space">Заблокированные</button>
            </div>
        )
    }
}

export default MobileCompanyFilters;