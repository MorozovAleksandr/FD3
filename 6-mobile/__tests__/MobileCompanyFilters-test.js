"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import MobileCompanyFilters from '../src/components/MobileCompany/MobileCompanyFilters/MobileCompanyFilters';

test('Рендер фильтра', () => {

    // создаём тестовую версию компонента
    const component = renderer.create(
        <MobileCompanyFilters />
    );

    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

});