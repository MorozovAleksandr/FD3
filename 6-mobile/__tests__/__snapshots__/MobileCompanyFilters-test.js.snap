// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Рендер фильтра 1`] = `
<div>
  <button
    className="btn btn-secondary btn-space"
    data-mode={0}
    onClick={[Function]}
    type="button"
  >
    Все
  </button>
  <button
    className="btn btn-secondary btn-space"
    data-mode={1}
    onClick={[Function]}
    type="button"
  >
    Активные
  </button>
  <button
    className="btn btn-secondary btn-space"
    data-mode={2}
    onClick={[Function]}
    type="button"
  >
    Заблокированные
  </button>
</div>
`;
